File: foxy/parent/src/command/vanilla/util/HelpExecutor.ts
import { bot } from "../../../FoxyLauncher";
import { colors } from "../../../../../../common/utils/colors";
import { createEmbed } from "../../../utils/discord/Embed";
import { ExecutorParams } from "../../structures/CommandExecutor";

export default class HelpExecutor {
    async execute({ context, endCommand, t }: ExecutorParams) {
        const embed = createEmbed({
            title: bot.helpers.foxy.getEmojiById(bot.emotes.FOXY_HOWDY) + " " + "Foxy",
            color: colors.FOXY_DEFAULT,
            description: t('commands:help.bot.description', { user: (await context.getAuthor()).asMention }),
            fields: [
                {
                    name: bot.helpers.foxy.getEmojiById(bot.emotes.FOXY_WOW) + " " + t('commands:help.bot.fields.addme'),
                    value: `[${t('help.bot.fields.add')}](${bot.foxy.constants.INVITE_LINK})`,
                },
                {
                    name: bot.helpers.foxy.getEmojiById(bot.emotes.FOXY_CUPCAKE) + " " + t('commands:help.bot.fields.support'),
                    value: bot.foxy.constants.SUPPORT_SERVER,
                },
                {
                    name: bot.helpers.foxy.getEmojiById(bot.emotes.FOXY_PRAY) + " " + t('commands:help.bot.fields.crowdin'),
                    value: bot.foxy.constants.CROWDIN,
                },
                {
                    name: bot.helpers.foxy.getEmojiById(bot.emotes.FOXY_SUNGLASSES) + " " + t('commands:help.bot.fields.website'),
                    value: bot.foxy.constants.FOXY_WEBSITE,
                },
                {
                    name: bot.helpers.foxy.getEmojiById(bot.emotes.FOXY_WOW) + " " + t('commands:help.bot.fields.privacy'),
                    value: bot.foxy.constants.TERMS,
                }
            ],
            thumbnail: {
                url: bot.foxy.constants.FOXY_AVATAR
            }
        });
        context.reply({ embeds: [embed] });
        return endCommand();
    }
}

File: foxy/src/main/kotlin/net/cakeyfox/foxy/command/vanilla/utils/HelpExecutor.kt
package net.cakeyfox.foxy.command.vanilla.utils

import net.cakeyfox.common.Colors
import net.cakeyfox.common.Constants
import net.cakeyfox.common.FoxyEmotes
import net.cakeyfox.foxy.command.FoxyInteractionContext
import net.cakeyfox.foxy.command.structure.FoxyCommandExecutor
import net.cakeyfox.foxy.utils.pretty

class HelpExecutor : FoxyCommandExecutor() {
    override suspend fun execute(context: FoxyInteractionContext) {
        context.reply {
            embed {
                description = pretty(
                    FoxyEmotes.FoxyHowdy,
                    context.locale["help.description", context.user.asMention]
                )

                color = Colors.FOXY_DEFAULT
                thumbnail = context.foxy.selfUser.effectiveAvatarUrl

                field {
                    name = context.locale["help.field.addMe", FoxyEmotes.FoxyWow]
                    value = "[${context.locale["help.field.addMeValue"]}](${Constants.INVITE_LINK})"
                    inline = false
                }

                field {
                    name = context.locale["help.field.support", FoxyEmotes.FoxyHug]
                    value = Constants.SUPPORT_SERVER
                    inline = false
                }

                field {
                    name = context.locale["help.field.website", FoxyEmotes.FoxyCake]
                    value = Constants.FOXY_WEBSITE
                    inline = false
                }

                field {
                    name = context.locale["help.field.terms", FoxyEmotes.FoxyRage]
                    value = Constants.TERMS
                    inline = false
                }
            }
        }
    }
}